bundles/org.openhab.binding.bluetooth.airthings/src/main/java/org/openhab/binding/bluetooth/airthings/internal/AirthingsWavePlusHandler.java

```diff
@@ -0,0 +1,213 @@
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+@NonNullByDefault
+public class AirthingsWavePlusHandler extends BeaconBluetoothHandler {
+
+    private static final String DATA_UUID = "b42e2a68-ade7-11e4-89d3-123b93f75cba";
+
+    private final Logger logger = LoggerFactory.getLogger(AirthingsWavePlusHandler.class);
+    private final UUID uuid = UUID.fromString(DATA_UUID);
+
+    private volatile Boolean servicesResolved = false;
+    private Optional<AirthingsConfiguration> configuration = Optional.empty();
+    private @Nullable ScheduledFuture<?> scheduledTask;
+
+    public AirthingsWavePlusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        configuration = Optional.of(getConfigAs(AirthingsConfiguration.class));
+        logger.debug("Using configuration: {}", configuration);
+    }
+
+    @Override
+    public void dispose() {
+        try {
+            super.dispose();
+        } finally {
+            if (scheduledTask != null) {
+                scheduledTask.cancel(true);
+                scheduledTask = null;
+            }
+            servicesResolved = false;
+        }
+    }
+
+    private void startScheduledTask() {
+        if (scheduledTask == null) {
+            configuration.ifPresent(cfg -> {
+                logger.debug("Start scheduled task to read device in every {} seconds", cfg.refreshInterval);
+                scheduledTask = scheduler.scheduleWithFixedDelay(this::execute, 10, cfg.refreshInterval,
+                        TimeUnit.SECONDS);
+            });
+        }
+    }
+
+    private void execute() {
+        ConnectionState connectionState = device.getConnectionState();
+        logger.debug("Device {} state is {}", address, connectionState);
+        switch (connectionState) {
+            case DISCONNECTED:
+                connect();
+                break;
+            case CONNECTED:
+                synchronized (servicesResolved) {
+                    if (servicesResolved) {
+                        read();
+                    } else {
+                        discoverServices();
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void connect() {
+        logger.debug("Connect to device {}...", address);
+        if (!device.connect()) {
+            logger.debug("Connecting to device {} failed", address);
+        }
+    }
+
+    private void disconnect() {
+        logger.debug("Disconnect from device {}...", address);
+        if (!device.disconnect()) {
+            logger.debug("Disconnect from device {} failed", address);
+        }
+    }
+
+    private void discoverServices() {
+        logger.debug("Discover services for device {}", address);
+        if (!device.discoverServices()) {
+            logger.debug("Discovering services failed");
+            disconnect();
+        }
```

Comment:
In the Bluez binding, there is no way initiate a service discovery manually, instead happens automatically when a device is connected to. As such, the `discoverServices` implementation always returns false. So you cant rely on a return of `false` as meaning something failed.

Reply:
Will Bluez send event when services are discovered (onServicesDiscovered)?

Reply:
Yep, in Bluez `onServicesDiscovered` will always get called after a successful connection (assuming the services get discovered properly)